_commands/tools/diff.md

```diff
@@ -2,23 +2,47 @@
 ---
 
 diff
--------
-TODO: Add documentation for this command by submitting a pull request.
-<!-- one line explanation would go here -->
+--
+
+`diff` is used to check for the differences between two files line by line. This is very useful for checking a programs output against a file containing its expected output.
 
-<!-- minimal example -->
 ~~~ bash
-*short* example of command usage and output
+$ printf "this\ndifference\n" > first.txt
+$ printf "this\nsimilarity\n" > second.txt
+$ diff first.txt second.txt
+2c2
+< difference
+---
+> similarity
+$
 ~~~
 
 <!--more-->
 
 ### Useful Options / Examples
 
-#### Example command
+#### Example Command
+When the files completely match, there is no output from `diff`.
+
+~~~ bash
+$ printf "same" > first.txt
+$ printf "same" > second.txt
+$ diff first.txt second.txt
+$
+~~~
 
-##### Break it down
 
-#### Example command
+#### `diff -y`
+
+This outputs both files in 2 columns with a '|' marking lines with differences, similar to the standard output of 'sdiff'.
+
+~~~ bash
+$ printf "working with github\nlets you share your work\ncollaboration" > first.txt
+$ printf "working with github\nlets you backup your data\nrefrigerator" > second.txt
+$ diff -y haiku1.txt haiku2.txt
```

Comment:
oh and the two printf's redirect into files called "first.txt" and "second.txt" yet the diff command uses "haiku1.txt" and "haiku2.txt"